File: test_decrypt.js
const fs = require('fs');
const CryptoJS = require('crypto-js');

const SECRET_KEY = 'vA$c1n_S3cr3t_K3y_!2024';

fetch('http://localhost:3000/api/proxy?endpoint=transcoddedFiles/id/3103838/')
  .then(res => res.json())
  .then(data => {
    const bytes = CryptoJS.AES.decrypt(data.payload, SECRET_KEY);
    const decryptedStr = bytes.toString(CryptoJS.enc.Utf8);
    console.log(JSON.stringify(JSON.parse(decryptedStr), null, 2));
  })
  .catch(console.error);

File: test_decrypt.py
import requests
import json
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad

def decrypt_payload(payload_b64):
    key = b'18261320293026211231313328221526'
    iv = b'1826132029302621'
    cipher = AES.new(key, AES.MODE_CBC, iv)
    decrypted = unpad(cipher.decrypt(base64.b64decode(payload_b64)), AES.block_size)
    return json.loads(decrypted.decode('utf-8'))

r = requests.get('https://alex-cinema.vercel.app/api/proxy?endpoint=video/1336019')
data = r.json()
payload = data.get('payload')
if payload:
    decrypted = decrypt_payload(payload)
    print("STREAMS:", json.dumps(decrypted.get('streams', []), indent=2))
else:
    print("No payload")
